compiler/src/iree/compiler/Codegen/Common/CPU/CPULowerToUKernels.cpp

```diff
@@ -534,8 +535,46 @@ matchDAGForUKernel(RewriterBase &rewriter, IREE::Codegen::QueryTileSizesOp op,
   if (tensorType.getRank() != 2) {
     return rewriter.notifyMatchFailure(op, "only the 2D case is implemented");
   }
-  auto encoding =
-      dyn_cast_or_null<IREE::Encoding::EncodingAttr>(tensorType.getEncoding());
+  Attribute tensorEncoding = tensorType.getEncoding();
+  if (!tensorEncoding) {
+    return rewriter.notifyMatchFailure(op,
+                                       "tensorType does not have encodings");
+  }
+  IREE::Encoding::EncodingAttr encoding;
+  if (auto encodingAttr =
+          dyn_cast<IREE::Encoding::EncodingAttr>(tensorEncoding)) {
+    encoding = encodingAttr;
+  } else if (auto layoutAttr =
+                 dyn_cast<IREE::Encoding::LayoutAttr>(tensorEncoding)) {
+    if (!llvm::all_of(layoutAttr.getLayouts().getValue(),
+                      llvm::IsaPred<IREE::CPU::VMVXEncodingLayoutAttr>)) {
+      return rewriter.notifyMatchFailure(
+          op, "only VMVX encoding resolver is handled");
+    }
+    for (auto attr : layoutAttr.getLayouts()
```

Comment:
Isn't this just looping through all layout attributes and overriding `encoding` every time? Can't we just get the last one or check and ensure that there is only a single layout entry?